integration_test/tests/advertisingTest.ts

```diff
@@ -153,29 +158,60 @@ export default (spec: TestScope) => {
     'progressive'
   );
 
-  function advertisingErrorTests(adItem: AdItem, label: string) {
+  function advertisingErrorTests(adItem: () => AdItem, label: string) {
     spec.describe(`scheduling erroneous ${label} ad`, () => {
       spec.it('emits AdError events', async () => {
         await startPlayerTest({}, async () => {
           await loadSourceConfig(Sources.artOfMotionHls);
           await callPlayerAndExpectEvent((player) => {
-            player.scheduleAd(adItem);
+            player.scheduleAd(adItem());
+            player.play();
+          }, EventType.AdError);
+        });
+      });
+
+      spec.it('validates AdError event properties', async () => {
+        await startPlayerTest({}, async () => {
+          await loadSourceConfig(Sources.artOfMotionHls);
+          const adErrorEvent = await callPlayerAndExpectEvent((player) => {
+            player.scheduleAd(adItem());
             player.play();
           }, EventType.AdError);
+
+          expect(
+            adErrorEvent.name,
+            'AdError event should have name'
+          ).toBeDefined();
+          expect(
+            adErrorEvent.name,
+            'AdError event name should be onAdError'
+          ).toBe('onAdError');
+          expect(
+            adErrorEvent.timestamp,
+            'AdError event should have timestamp'
+          ).toBeDefined();
+          expect(
+            typeof adErrorEvent.timestamp,
+            'timestamp should be a number'
+          ).toBe('number');
+          expect(
+            adErrorEvent.timestamp,
+            'timestamp should be > 0'
+          ).toBeGreaterThan(0);
         });
       });
     });
   }
 
-  advertisingErrorTests(
-    { sources: [{ tag: AdTags.error, type: AdSourceType.IMA }] },
-    'IMA'
-  );
+  advertisingErrorTests(() => {
```

Comment:
Changed to always fetch new tags with new correlator to avoid IMA not playing the same URL